pkg/sql/plan/base_binder.go

```diff
@@ -904,6 +905,24 @@ func bindFuncExprImplByPlanExpr(name string, args []*Expr) (*plan.Expr, error) {
 		if int(args[0].Typ.Id) != int(args[1].Typ.Id) {
 			return nil, moerr.NewInvalidInput(name + " function have invalid input args type")
 		}
+	case "str_to_date", "to_date":
+		if len(args) != 2 {
+			return nil, moerr.NewInvalidArg(name+" function have invalid input args length", len(args))
+		}
+
+		if args[1].Typ.Id == int32(types.T_varchar) || args[1].Typ.Id == int32(types.T_char) {
+			if exprC, ok := args[1].Expr.(*plan.Expr_C); ok {
+				sval := exprC.C.Value.(*plan.Const_Sval)
+				tp, _ := binary.JudgmentToDateReturnType(sval.Sval)
+				args = append(args, makePlan2DateConstNullExpr(tp))
```

Comment:
I don't think you use NullExpr is a good choice. Null is null.

May be you can use  Int64Expr， different value for different type.  like:  1=Date 2=Datetime 3=...